Update how to delete all resources in cattle-monitoring-system (#495)
harvester/docs: docs/upgrade/v1-1-2-to-v1-2-0.md

- Related issue
  - [[BUG] The rancher-monitoring Pod stuck at terminating status when upgrading from v1.1.2 to v1.2.0-rc6](https://github.com/harvester/harvester/issues/4484)

> #### Many PODs in cattle-monitoring-system namespace are to be deleted

Reviewer: ```suggestion
#### Multiple pods in the cattle-monitoring-system namespace must be deleted
```

Author: Try to keep format:
```
#### POD prometheus-rancher-monitoring-prometheus-0 is to be deleted
#### Multiple PODs in cattle-monitoring-system namespace are to be deleted
```

> 1. Check the log of the `apply-manifests` pod to see if the following messages repeat.

Reviewer: ```suggestion
1. Check if the following message is repeatedly displayed in the `apply-manifests` pod logs.
```

Author: Keep same as L293  | L344 
```
1. Check the log of the `apply-manifests` pod to see if the following messages repeat.
```

    ```
    there are still 10 pods in cattle-monitoring-system to be deleted
    Fri Dec  8 19:06:56 UTC 2023
    there are still 10 pods in cattle-monitoring-system to be deleted
    Fri Dec  8 19:07:01 UTC 2023
    ```

>     When it continues to show 10 (or other number) pods, it encounters below issue.
> 
> 1. The root cause:

Reviewer: ```suggestion

    When the repeated messages show the same number of pods, the following issue has occurred.
```

Reviewer: Displaying the root cause does not involve any user action so it should not be presented as a step in the procedure.

    ```
    The monitoring feature is deployed from the rancher-monitoring ManagedChart, in Harvester v1.2.0,v1.2.1,
    this ManagedChart is converted to Harvester Addon feature when upgrading.
    The ManagedChart rancher-monitoring is deleted, normally, all the generated resources including deployment, 
    daemonset etc. will be deleted automatically. But in this case, those resources are not deleted.
    The above log reflects the result.
    Following instructions will guide to delete them manually.
    ```

> 1. The resources in cattle-monitoring-system:

Reviewer: ```suggestion
1. Locate the affected resources in the `cattle-monitoring-system` namespace.
```

    ```
    Root level resources in cattle-monitoring-system

    Customized CRD: Prometheus
      Object: rancher-monitoring-prometheus
      Sub-object: statefulset.apps/prometheus-rancher-monitoring-prometheus

    Customized CRD: Alertmanager
      object:  rancher-monitoring-alertmanager 
      Sub-object:  statefulset.apps/alertmanager-rancher-monitoring-alertmanager

    Deployment:
      rancher-monitoring-grafana
      rancher-monitoring-kube-state-metrics
      rancher-monitoring-operator
      rancher-monitoring-prometheus-adapter

    Daemonset:
      rancher-monitoring-prometheus-node-exporter
    ```

> 1. Delete those resources manually:

Reviewer: ```suggestion
1. Delete the affected resources.
```

    ```
    Use below commands to delete them, meanwhile check the log of the `apply-manifests` until it does not
    report `there are still x pods in cattle-monitoring-system to be deleted`.

    kubectl delete prometheus rancher-monitoring-prometheus -n cattle-monitoring-system
    kubectl delete alertmanager rancher-monitoring-alertmanager -n cattle-monitoring-system

    kubectl delete deployment rancher-monitoring-grafana -n cattle-monitoring-system
    kubectl delete deployment rancher-monitoring-kube-state-metrics -n cattle-monitoring-system
    kubectl delete deployment rancher-monitoring-operator -n cattle-monitoring-system
    kubectl delete deployment rancher-monitoring-prometheus-adapter -n cattle-monitoring-system

    kubectl delete daemonset rancher-monitoring-prometheus-node-exporter -n cattle-monitoring-system
    ```
    
    :::note

>     Some of above commands may need to run several times to delete the resources finally.

Reviewer: ```suggestion
    You may need to run some of the commands more than once to completely delete the resources.
```
